.github/workflows/python-checks.yml

```diff
@@ -27,27 +27,30 @@ jobs:
             - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
 
             - name: Set up Python ${{ matrix.python-version }}
-              uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
```

Comment:
what about letting this float, and then we can have the the dest python be "3.14-dev" which will translate to whatever prerelease is most recently supported by actions/setup-python?

Reply:
I am against unpinning actions.
I might be open to compromising to using the tags, like V5, but hash pinning is recommended for GitHub Actions.

Reply:
great, actions/setup-python@v5 would be perfect!